Server/core/CCrashHandler.cpp

```diff
@@ -212,103 +209,103 @@ void CCrashHandler::DumpMiniDump(_EXCEPTION_POINTERS* pException, CExceptionInfo
-                    pExceptionInformation->GetSS(), pExceptionInformation->GetES(), pExceptionInformation->GetFS(), pExceptionInformation->GetGS());
-
-                fprintf(pFile, "%s", strInfo.c_str());
-
-                // End of unhandled exception
-                fprintf(pFile, "%s", "** -- End of unhandled exception -- **\n\n\n");
-
-                // Close the file
-                fclose(pFile);
-            }
+            // Close the dumpfile
+            CloseHandle(hFile);
+
+            FileSave(PathJoin(ms_strDumpPath, "server_pending_upload_filename"), strFinalDumpPathFilename);
         }
 
-        // Free the DLL again
-        FreeLibrary(hDll);
+        // Write a log with the generic exception information
+        FILE* pFile = File::Fopen(PathJoin(ms_strDumpPath, "server_pending_upload.log"), "a+");
+        if (pFile)
+        {
+            // Header
+            fprintf(pFile, "%s", "** -- Unhandled exception -- **\n\n");
+
+            // Write the time
+            time_t timeTemp;
+            time(&timeTemp);
+
+            SString strMTAVersionFull = SString("%s", MTA_DM_BUILDTAG_LONG);
+
+            SString strInfo;
+            strInfo += SString("Version = %s\n", strMTAVersionFull.c_str());
+            strInfo += SString("Time = %s", ctime(&timeTemp));
+
+            strInfo += SString("Module = %s\n", pExceptionInformation->GetModulePathName());
+
+            // Write the basic exception information
+            strInfo += SString("Code = 0x%08X\n", pExceptionInformation->GetCode());
+            strInfo += SString("Offset = 0x%08X\n\n", pExceptionInformation->GetAddressModuleOffset());
+
+            // Write the register info
+            strInfo += SString(
+                "EAX=%08X  EBX=%08X  ECX=%08X  EDX=%08X  ESI=%08X\n"
+                "EDI=%08X  EBP=%08X  ESP=%08X  EIP=%08X  FLG=%08X\n"
+                "CS=%04X   DS=%04X  SS=%04X  ES=%04X   "
+                "FS=%04X  GS=%04X\n\n",
+                pExceptionInformation->GetEAX(), pExceptionInformation->GetEBX(), pExceptionInformation->GetECX(), pExceptionInformation->GetEDX(),
+                pExceptionInformation->GetESI(), pExceptionInformation->GetEDI(), pExceptionInformation->GetEBP(), pExceptionInformation->GetESP(),
+                pExceptionInformation->GetEIP(), pExceptionInformation->GetEFlags(), pExceptionInformation->GetCS(), pExceptionInformation->GetDS(),
+                pExceptionInformation->GetSS(), pExceptionInformation->GetES(), pExceptionInformation->GetFS(), pExceptionInformation->GetGS());
+
+            fprintf(pFile, "%s", strInfo.c_str());
+
+            // End of unhandled exception
+            fprintf(pFile, "%s", "** -- End of unhandled exception -- **\n\n\n");
+
+            // Close the file
+            fclose(pFile);
+        }
     }
+
+    // Free the DLL again
+    FreeLibrary(hDll);
```

Comment:
you could also just wrap `hDll` into an `unique_ptr`.
That way you can add even more early outs